int32 center_x
int32 center_y
float64 data
float64 angle
